public/components/ui_action_context_menu.tsx

```diff
@@ -75,14 +107,13 @@ export const ActionContextMenu = (props: Props) => {
     [actionContext.datasetId, actionContext.datasetType, actionContext.dataSourceId]
   );
 
-  // The action button should be not displayed when there is no action and result summary disabled.
-  if (actionsRef.current.length === 0 && !resultSummaryEnabled) {
+  // The action button should be not displayed when there is no action and result summary disabled or there is no data2Summary agent
+  if (!isSummaryAgentAvailable || (actionsRef.current.length === 0 && !resultSummaryEnabled)) {
```

Comment:
I'd prefer to move isSummaryAgentAvailable behind !resultSummaryEnabled because we need to make a network request to get the status.